### PR TITLE
Features/reminders_notification_improve

### PR DESCRIPTION
**Pull request by natgate**
With this commit: 
- the description of the notifications for the reminders is more precise, it adds a precision of time left like for example : "expires in 1 week" and a precision of which document expires for example: "Your salary slips expires..."
- notifications are shown for all documents
- when changing a date of expiration, it automatically remove the previously scheduled notifications

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java [hunk 1]
```diff
@@ -166,57 +169,85 @@ private void createNotificationChannel() {
     }
 
 
-    private void scheduleNotification(int dayOfMonth, int month, int year) {
+    private void clearPreviouslyScheduledNotif() {
 
+        int all_reminders_size = RemindersDate.values().length;
```
Comment (Scott2Williams) [inline app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java:174]: I believe the only place we use snake_cased variables is for UI components accessed via R.id.<component_id>. Consider camelCase e.g. `allRemindersSize` or let me know if I'm simply wrong.

File: app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java [hunk 2]
```diff
@@ -166,57 +169,85 @@ private void createNotificationChannel() {
...
+                scheduledMessage.add(period.message);
+            }
 
-        return scheduleTime;
+        }
 
+        Calendar nowTestDemo = Calendar.getInstance();
+        nowTestDemo.add(Calendar.SECOND, 20);
+        scheduledTime.add(nowTestDemo.getTimeInMillis());
+        scheduledMessage.add("in 2 days");
```
Comment (Scott2Williams) [inline app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java:250]: This looks like some test code that made it into the final commit. Maybe remove it if it's not needed.

**Review by Scott2Williams — APPROVED** [review]
Good stuff. I think the granularity of time between reminders that you chose is appropriate.

**Review by paul-ndl — APPROVED** [review]
nice feature! you can maybe add notification title to Constants class to avoid code repetition

File: app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java (hunk 2 above)
Comment (natgate) [reply app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java:250]: Yeah it is for the demo but in the final version we will remove that!

File: app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java (hunk 1 above)
Comment (natgate) [reply app/src/main/java/ch/epfl/sweng/hostme/wallet/ExpirationDatePicker.java:174]: Thanks for noticing that, I'll change it!

**Pull request merged**